docker_daemon/check.py

```diff
@@ -425,6 +432,19 @@ def _get_tags(self, entity=None, tag_type=None):
                     ecs_tags = self.ecs_tags[entity_id]
                     tags.extend(ecs_tags)
 
+            # Add Mesos tags
+            if self.collect_mesos_tags and (tag_type is CONTAINER or tag_type is PERFORMANCE):
+                cont_id = entity.get("Id")
+                if cont_id is not None:
+                    try:
+                        cont_inspected = self.docker_client.inspect_container(cont_id)
```

Comment:
`docker inspect` can be very slow. We avoid as much as possible to use it, and when we do we cache the result.
In this case I think it's legit to call it, but could you cache the tags to avoid calling it at each run? I know it's more work because we don't yet provide a generic way to do it - sorry about that.
You can have a look at https://github.com/DataDog/integrations-core/pull/305 and https://github.com/DataDog/dd-agent/pull/3289 that implement the same logic but for Nomad.

If you could create a MesosUtil similar to NomadUtil and that would only implement the `extract_container_tags` method it would be good enough. We can handle the rest of the wiring.
Ping me if you need any pointer.